src/libexpr/primops/fetchMercurial.cc

```diff
@@ -204,6 +203,11 @@ static void prim_fetchMercurial(EvalState & state, const Pos & pos, Value * * ar
         state.allowedPaths->insert(hgInfo.storePath);
 }
 
-static RegisterPrimOp r("fetchMercurial", 1, prim_fetchMercurial);
+static void reg(EvalState & state)
+{
+    state.addPrimOp("fetchMercurial", 1, prim_fetchMercurial);
+}
+
+static RegisterPlugin r(Plugin(PluginTag<PluginType::EvalPlugin>(), reg));
```

Comment:
I don't understand why these changes are necessary. Doesn't the `RegisterPrimOp` constructor run automatically when the plugin is loaded? Why the need for a `RegisterPlugin` with a lot of template crap?

Reply:
It's not necessary, this is all contained in the second commit that I explicitly said could be dropped ;) This was just to drop duplicate capabilities.

Also, a single template is hardly "a lot of template crap".

Reply:
... Though I just realized an easy way to avoid the template altogether, patch incoming.

Reply:
Well, I don't understand the need for all this over-engineering (e.g. `Plugin(PluginTag<PluginType::EvalPlugin>()`), plus all that complexity in `plugins.hh`, which for some reason has a forward-decl to `EvalState`...

Reply:
I removed the tag stuff, was a holdover from an earlier approach. `plugins.hh` needs to reference `EvalState &` to definte the `EvalPlugin` type, but #including `eval.hh` this low in the header chain could easily lead to accidental references to libnixexpr in libnixstore, if it even worked otherwise.